app/src/main/java/de/westnordost/streetcomplete/quests/religion/AddReligionToWaysideShrineForm.java

```diff
@@ -0,0 +1,84 @@
+package de.westnordost.streetcomplete.quests.religion;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.quests.ImageListQuestAnswerFragment;
+import de.westnordost.streetcomplete.quests.PriorityList;
+import de.westnordost.streetcomplete.view.Item;
+
+public class AddReligionToWaysideShrineForm extends ImageListQuestAnswerFragment
+{
+    private static final int INITIALLY_DISPLAYED_ITEMS = 8;
+
+    private static final Item[] ALL_RELIGION_VALUES = new Item[]{
+            // worldwide usage, values covering vast majority of used tags
+            new Item("christian",      R.drawable.ic_religion_christian,      R.string.quest_religion_christian),
+            new Item("shinto",      R.drawable.ic_religion_shinto,      R.string.quest_religion_shinto),
+            new Item("buddhist",      R.drawable.ic_religion_buddhist,      R.string.quest_religion_buddhist),
+            new Item("bahai",      R.drawable.ic_religion_bahai,      R.string.quest_religion_bahai),
+            new Item("caodaism",      R.drawable.ic_religion_caodaist,      R.string.quest_religion_caodaist),
+            new Item("confucian",      R.drawable.ic_religion_confucian,      R.string.quest_religion_confucian),
+            new Item("hindu",      R.drawable.ic_religion_hindu,      R.string.quest_religion_hindu),
+            new Item("jain",      R.drawable.ic_religion_jain,      R.string.quest_religion_jain),
+            new Item("jewish",      R.drawable.ic_religion_jewish,      R.string.quest_religion_jewish),
+            new Item("muslim",      R.drawable.ic_religion_muslim,      R.string.quest_religion_muslim),
+            new Item("sikh",      R.drawable.ic_religion_sikh,      R.string.quest_religion_sikh),
+            new Item("taoist",      R.drawable.ic_religion_taoist,      R.string.quest_religion_taoist),
+    };
+
+    private Item[] actualReligionsValues;
+
+    @Override public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                                       Bundle savedInstanceState)
+    {
+        View view = super.onCreateView(inflater, container, savedInstanceState);
+        actualReligionsValues = createItems();
+        imageSelector.setCellLayout(R.layout.cell_icon_select_with_label_below);
+
+        addOtherAnswer(R.string.quest_religion_for_place_of_worship_answer_multi, this::applyMultiAnswer);
```

Comment:
Does this really exist?

Reply:
Probably the real question is whatever it is popular enough to keep it, as in case of religion basically anything imaginable exists somewhere.

In this case it was more work to throw it away than to keep it - I will not defend it if removal makes code maintenance easier or is desirable for some other reason.